internal/core/service/user/service.go

```diff
@@ -12,17 +13,19 @@ import (
 
 type Service interface {
 	RegisterUser(ctx context.Context, req request.RegisterRequestBody) (user entity.User, err error)
-	LoginUser(ctx context.Context, req request.LoginRequestBody) (user entity.User, err error)
+	LoginUser(ctx context.Context, req request.LoginRequestBody) (user entity.User, accessToken string, err error)
 	GetListUser(ctx context.Context, limit int, page int, role string) (users []entity.User, err error)
 }
 
 type service struct {
-	storage UserStorage
+	storage    UserStorage
+	encryption encryption.JWTService
 }
 
-func NewService(storage UserStorage) Service {
+func NewService(storage UserStorage, jwtService encryption.JWTService) Service {
```

Comment:
Use a custom config pattern to initialize the object. It gives a lot of flexibility such as: backward compatibility, cleaner code. Ref: https://golang.cafe/blog/golang-functional-options-pattern.html


Also always give validation checking during initialization to make sure the required configs is filled properly. Check [this](https://github.com/Haraj-backend/hex-monscape/blob/master/internal/driver/rest/api.go#L33) as ref.